components/ReviewModal.tsx

```diff
@@ -1,24 +1,73 @@
 import Slider from "@react-native-community/slider";
 import { useState } from "react";
 import { Alert, Modal, Text, Pressable, View, TextInput } from "react-native";
+import { postReview } from "../utils/api";
+import { useGlobalSearchParams } from "expo-router";
+import { useUserData } from "../app/contexts/UserContent";
+import { Review } from "../types/front-end";
+import { AntDesign } from "@expo/vector-icons";
 
-const ReviewModal = () => {
+interface Iprops {
+  setReviews: Function;
+}
+
+const ReviewModal = (props: Iprops) => {
   const [modalVisible, setModalVisible] = useState(false);
   const [title, setTitle] = useState("");
   const [body, setBody] = useState("");
   const [rating, setRating] = useState(0);
 
+  const { music_id } = useGlobalSearchParams();
+
+  const { user } = useUserData();
+
   const handleTitle = (input: string) => {
     setTitle(input);
   };
   const handleBody = (input: string) => {
     setBody(input);
   };
-  const handleRate = (input: number) => {};
-  
-  const handleSubmit = () => {
-    // TODO add util function to post user
+
+  const handleSubmit = async () => {
+    if (title && body && rating) {
+      try {
+        const postedReview = await postReview(music_id as string, {
+          screen_name: user.username,
+          rating: rating,
+          review_title: title,
+          review_body: body,
+        });
+
+        setModalVisible(!modalVisible);
+        setRating(0);
+
+        props.setReviews((currentReviews: Review[]) => {
+          return [
+            {
+              screen_name: user.username,
+              rating: rating,
+              review_title: title,
+              review_body: body,
+              created_at: new Date().toISOString(),
+            },
+            ...currentReviews,
+          ];
+        });
```

Comment:
I'm stealing this next time I need to optimistically render something and can't be bothered to type and there's nothing you can do about it